build-tools/automation/azure-pipelines-oss.yaml

```diff
@@ -3,65 +3,188 @@
+        version: $(DotNetCorePreviewVersion)
+
+    - template: yaml-templates/use-dot-net.yaml
+      parameters:
+        version: $(DotNetCoreVersion)
+
+    # Prepare and build everything
+    - script: |
+        echo "make prepare-update-mono V=1 CONFIGURATION=$(XA.Build.Configuration) $(PREPARE_FLAGS) MSBUILD_ARGS='$(EXTRA_MSBUILD_ARGS)'"
+        make prepare-update-mono V=1 CONFIGURATION=$(XA.Build.Configuration) $(PREPARE_FLAGS) MSBUILD_ARGS='$(EXTRA_MSBUILD_ARGS)'
+      workingDirectory: $(Build.SourcesDirectory)
+      displayName: make prepare-update-mono
+
+    - script: |
+        echo "make prepare CONFIGURATION=$(XA.Build.Configuration) V=1 $(PREPARE_FLAGS) MSBUILD_ARGS='$(EXTRA_MSBUILD_ARGS)'"
+        make prepare CONFIGURATION=$(XA.Build.Configuration) V=1 $(PREPARE_FLAGS) MSBUILD_ARGS='$(EXTRA_MSBUILD_ARGS)'
+      workingDirectory: $(Build.SourcesDirectory)
+      displayName: make prepare
+
+    - script: |
+        echo "make jenkins V=1 CONFIGURATION=$(XA.Build.Configuration) PREPARE_CI=1 PREPARE_AUTOPROVISION=1"
+        make jenkins V=1 CONFIGURATION=$(XA.Build.Configuration) PREPARE_CI=1 PREPARE_AUTOPROVISION=1
+      workingDirectory: $(Build.SourcesDirectory)
+      displayName: make jenkins
+
+    - script: |
+        echo "make create-installers V=1 CONFIGURATION=$(XA.Build.Configuration)"
+        make create-installers V=1 CONFIGURATION=$(XA.Build.Configuration)
+      workingDirectory: $(Build.SourcesDirectory)
+      displayName: create installers
+
+    - script: |
+        echo "make package-oss CONFIGURATION=$(XA.Build.Configuration) V=1"
+        make package-oss CONFIGURATION=$(XA.Build.Configuration) V=1
+      workingDirectory: $(Build.SourcesDirectory)
+      displayName: package oss
+
+    - script: |
+        echo "all-tests CONFIGURATION=$(XA.Build.Configuration) V=1"
+        make all-tests CONFIGURATION=$(XA.Build.Configuration) V=1
+      workingDirectory: $(Build.SourcesDirectory)
+      displayName: make all-tests
+      condition: and(succeeded(), eq(variables['BUILD_TESTS'], 'true'))
+
+    - script: |
+        echo "make package-build-status CONFIGURATION=$(XA.Build.Configuration) V=1"
+        make package-build-status CONFIGURATION=$(XA.Build.Configuration) V=1
+      workingDirectory: $(Build.SourcesDirectory)
+      displayName: package build status
+
+    - task: MSBuild@1
+      displayName: pack all nupkgs
+      inputs:
+        solution: $(Build.SourcesDirectory)/build-tools/create-packs/Microsoft.Android.Sdk.proj
+        configuration: $(XA.Build.Configuration)
+        msbuildArguments: /t:CreateAllPacks /restore /bl:$(Build.SourcesDirectory)/bin/Build$(XA.Build.Configuration)/create-all-packs.binlog
+
+    - task: PublishPipelineArtifact@1
+      displayName: upload nupkgs
+      inputs:
+        artifactName: $(NuGetArtifactName)
+        targetPath: $(Build.SourcesDirectory)/bin/Build$(XA.Build.Configuration)/$(NuGetArtifactName)
+
+    - script: |
+        exitCode=0
```

Comment:
I don't understand why you need this pattern.  `script:` blocks will fail if the last command executed fails, so all that this `script:` block *should* need is:

```
- script: >
  make run-performance-tests CONFIGURATION=$(XA.Build.Configuration) V=1
```

If it fails, then the entire script step *also* fails.  There shouldn't be a need for everything else.

Reply:
Ah, yes. This was a literal translation of the Jenkins groovy script here: https://github.com/xamarin/xamarin-android/blob/3f438e46d7b166a3a3ef54c9ffafb5f426760468/build-tools/automation/build.groovy#L304-L306.  Was needed to ensure the Jenkins pipeline stage result was set correctly. In the Azure DevOps world, the same logic isn't needed anymore and we can just rely on the pass/fail of the command. Fixed with commit f6b997c39d8fa35aed2c5f44d0ca86c1ff7c82f8